web/src/components/tasks/NaicsCodeInput.tsx

```diff
@@ -141,9 +148,14 @@ export const NaicsCodeInput = (props: Props): ReactElement => {
               name={"naics-radio-group"}
               value={naicsCode}
               onChange={(event: React.ChangeEvent<{ name?: string; value: string }>): void => {
-                setNaicsCode(event.target.value);
                 if (displayInput) setDisplayInput(false);
-                setInProgress();
+                if (isAuthenticated === IsAuthenticated.TRUE) {
```

Comment:
I would also prefer to see this in a guard clause with the main logic of the function at the "top-level" instead of within an if statement